web/src/containers/ActivityDetailCostAllocateFFP.js

```diff
@@ -1,96 +1,105 @@
+                  label={t('activities.costAllocate.ffp.labels.other')}
+                  value={costAllocation[year].other}
+                  onChange={this.handleOther(year)}
                 />
-              ))}
-            </div>
-            {ffpLeft !== 0 && (
-              <div className="mt2 p1 h6 alert alert-error">
-                {ffpLeft < 0
-                  ? `You're over by ${Math.abs(ffpLeft)}%`
-                  : `Please allocate an additional ${ffpLeft}%`}
-              </div>
-            )}
-            <hr />
-            <div className="flex mxn-tiny">
-              {ffpData.map(({ id, amount }) => (
-                <div key={id} className="col-12">
-                  <div>{ffpDisplay[id]}</div>
-                  <div className="bold mono">{formatMoney(amount)}</div>
+                <div>{t('activities.costAllocate.ffp.totalNetOther')}</div>
+                <div className="h3 bold mono">{formatMoney(totalNetOther)}</div>
+                <hr />
+                <Select
+                  name={`ffp-${year}`}
+                  label={t('activities.costAllocate.ffp.labels.fedStateSplit')}
+                  options={['90-10', '75-25', '50-50']}
+                  onChange={this.handleFFP(year)}
+                />
+                <div className="flex mxn-tiny">
+                  {allocations.map(({ id, amount }) => (
+                    <div key={id} className="col-12">
+                      <div>{titleCase(id)}</div>
+                      <div className="h3 bold mono">{formatMoney(amount)}</div>
+                    </div>
+                  ))}
                 </div>
-              ))}
+              </div>
             </div>
-          </div>
+          ))}
         </div>
-      ))}
-    </div>
-  </div>
-);
+      </div>
+    );
+  }
+}
 
 ActivityDetailCostAllocateFFP.propTypes = {
   aId: PropTypes.number.isRequired,
   byYearData: PropTypes.array.isRequired,
+  costAllocation: PropTypes.object.isRequired,
   updateActivity: PropTypes.func.isRequired
 };
 
+// TODO [bren]: tidy up this data munging
 const mapStateToProps = ({ activities: { byId } }, { aId }) => {
   const activity = byId[aId];
+  const { costAllocation } = activity;
   const totals = getActivityTotals(activity);
 
   const byYearData = Object.keys(totals).map(year => {
     const total = totals[year];
-    const ffp = activity.costFFP[year];
+    const { ffp, other } = costAllocation[year];
+    const totalNetOther = total - other;
 
-    const ffpLeft = 100 - Object.values(ffp).reduce((a, b) => a + b, 0);
-    const ffpData = Object.entries(ffp).map(([id, percent]) => ({
+    const allocations = Object.keys(ffp).map(id => ({
       id,
-      percent,
-      amount: total * percent / 100
+      amount: totalNetOther * ffp[id] / 100
     }));
 
-    return { year, total, ffpLeft, ffpData };
+    return { year, total, totalNetOther, allocations };
   });
 
-  return { byYearData };
+  return { byYearData, costAllocation };
```

Comment:
As I'm reading this, one of the things I really like about redux and `mapStateToProps` is that I don't have to just *know* what these objects look like - I can go look at the appropriate reducer.  So nice.